invenio_vocabularies/datastreams/writers.py

```diff
@@ -53,15 +54,38 @@ def __init__(self, service_or_name, identity, *args, **kwargs):
 
         super().__init__(*args, **kwargs)
 
-    def write(self, stream_entry, *args, **kwargs):
-        """Writes the input entry using a given service."""
+    def _create(self, entry):
+        """Writes a new entry."""
+        try:
+            return self._service.create(self._identity, entry)
+        except PIDAlreadyExists:
+            raise WriterError(
+                [f"Vocabulary entry already exists: {entry}"]
+            )
+
+    def _update(self, entry):
+        """Updates an entry and creates if it does not exist."""
+        vocab_id = (entry["type"], entry["id"])
+        try:
+            current = self._service.read(self._identity, vocab_id)
+            updated = dict(current.to_dict(), **entry)
+            return self._service.update(self._identity, vocab_id, updated)
+        except PIDDoesNotExistError:
+            return self._service.create(self._identity, entry)
+
+    def write(self, stream_entry, force=False, *args, **kwargs):
```

Comment:
I agree with this. I opted by the `force` flag at a `write` level because it seemed more natural to me to give the choice per `record` instead of a `writer` arg (i.e. update all or create all). But it is true that given the use cases we have for `datastreams`/`writers` a writer arg works and pollute less the APIs